test/atm/ATMGovernanceProxyConstructorTest.js

```diff
@@ -0,0 +1,52 @@
+// JS Libraries
+const withData = require('leche').withData;
+const { t, encode } = require('../utils/consts');
+
+// Mock contracts
+const Mock = artifacts.require("./mock/util/Mock.sol");
+
+// Smart contracts
+const ATMGovernance = artifacts.require('./base/atm/ATMGovernance.sol');
+const ATMGovernanceProxy = artifacts.require('./base/atm/ATMGovernanceProxy.sol');
+
+contract('ATMGovernanceProxyConstructorTest', function (accounts) {
+    let atmGovernance
+    let atmSettings
+
+    beforeEach(async () => {
+        atmGovernance = await ATMGovernance.new();
+
+        atmSettings = await Mock.new();
+        await atmSettings.givenMethodReturnAddress(
+            encode(web3, 'atmGovernanceLogic()'),
+            atmGovernance.address
+        );
+    })
+
+    withData({
+        _1_initialize_basic: [accounts[0]],
+    }, function (
+        expectedErrorMessage,
+        mustFail
+    ) {
+        it(t('user', 'constructor', 'Should be able to create and initialize contract.', mustFail), async function() {
+            // Setup
+            try {
+                const proxy = await ATMGovernanceProxy.new(atmSettings.address)
```

Comment:
Are we testing this invocation?
Should we add assert(!mustFail) after the invocation?

Reply:
It is being tested by checking if the constructor of the proxy contract called the `initialize` function of the implementation and set the passed in `atmSettings` address.